documentation/projects/proposals/search_relevancy_sandbox/20230530-implementation_plan_staging_elasticsearch_reindex_dags.md

```diff
@@ -0,0 +1,192 @@
+# 2023-06-08 Implementation Plan: Staging Elasticsearch Reindex DAGs
+
+**Author**: @krysal
+
+## Reviewers
+
+- [ ] @AetherUnbound
+- [ ] @sarayourfriend
+
+## Project links
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/392)
+- [Project Proposal](./20230331-project_proposal_search_relevancy_sandbox.md)
+
+## Overview
+
+This document describes the addition of two DAGs for Elasticsearch (ES) index
+creation ––full and proportional-by-provider–– which will allow us to decouple
+the process from the long Ingestion server's data refresh process and experiment
+with smaller indices. Also includes the adoption of two new index aliases for
+ease of handling and querying the new index types from the API with the
+[`internal__index`](api_ii_param) param.
+
+[api_ii_param]: https://github.com/WordPress/openverse/pull/2073
+
+## Expected Outcomes
+
+- 1 DAG for fully recreation of the index that the API uses which includes the
+  includes all the database contents.
+- 1 DAG to create indexes of reduced size but
+  proportional-to-production-by-provider index.
+
+## Dependencies
+
+Same as for
+[Implementation Plan: Update Staging Database](./20230406-implementation_plan_update_staging_database.md).
+
+This work is related to the
+[Staging database recreation DAG](staging_db_recreation) plan for having
+production volumes in the staging DB, which is expected to finish soon.
+
+[staging_db_recreation]: https://github.com/WordPress/openverse/issues/1989
+
+## DAGs
+
+### `recreate_full_<media>_index` DAG
+
+#### Parameters
+
+1. `media_type`: The media type for which the index is being created. Presently
+   this would only be `image` or `audio`.
+2. `promote`: (Optional) A boolean value to indicate if result index will
+   replace the current one pointed by the `media` alias. If `True`, then the new
+   index will be the one used by the staging API. Defaults to `False`.
+3. `delete_old_if_promoted`: (Optional) A boolean value to indicate if the old
+   index pointed by the `<media>` alias should be deleted after replacement.
+   Defaults to `False`.
+
+#### Outlined Steps
+
+This DAG will leverage the **Ingestion server's API** and use the existing
+[`REINDEX` task](REINDEX), which is the same used by the data refresh process to
+create the index.
+
+1. Get the current timestamp in order to create the index suffix with the form
+   of `full-<timestamp>`.
+2. Use the `ingestion_server.trigger_and_wait_for_task()` utility to send the
+   `REINDEX` call to ingestion server, passing the previously generated
+   `index_suffix` in the data payload.
+3. If `promoted=True` is passed, then inmediatly make the `media` alias point to
+   the new index. A
```

Comment:
Agreed, I think we might want to use `POINT_ALIAS` terminology here instead.

Reply:
Hm, I thouht it would be better differentiate it from the `point_alias` task of the Ingestion server because it involves deleting the old index, which is not the case here. However, the promote task (from Ingestion server) also involves that so makes sense to use the more obvious name for this param 👍 Noted.